system-tests/lib/cre/environment/dons.go

```diff
@@ -56,6 +56,7 @@ func StartDONs(
 	capabilityConfigs cre.CapabilityConfigs,
 	copyCapabilityBinaries bool,
 	nodeSets []*cre.NodeSet,
+	apiUser, apiPassword string,
```

Comment:
I'd rather have a API user/password provider that would return this or that value per each infra type and inject that here than have user/password fields that are only used in case of k8s. It will be confusing for users.